File: JavaScript/TwoSum.js

target = 9
nums = [2,7,11,15]

var twoSum = function(nums, target) {
    var a = {} 
    for(var i = 0; i < nums.length; i++){
        var dif = target - nums[i] 
        if(a[dif] !== undefined){
            return [a[dif], i];
        }
        a[nums[i]] = i
    } 
};

console.log(twoSum(nums, target));

File: Python/TwoSum.py
target = 9
nums = [2,7,11,15]

def twoSum(nums, target):
    a = {}
    i = 0  
    for i in range(len(nums)):
        dif = target - nums[i] 
        if dif in a:
            return [a[dif], i];
        
        a[nums[i]] = i
    

print(twoSum(nums, target));